core/blockchain_test.go

```diff
@@ -4326,3 +4332,75 @@ func TestEIP3651(t *testing.T) {
 		t.Fatalf("sender balance incorrect: expected %d, got %d", expected, actual)
 	}
 }
+
+func TestRewindOnConfigChange(t *testing.T) {
+	genesisTime := uint64(12)
+
+	type testCase struct {
+		name              string
+		override1         func(*params.ChainConfig)
+		override2         func(*params.ChainConfig)
+		expectChainRewind bool
+	}
+
+	tcs := []testCase{
+		{
+			name:              fmt.Sprintf("CanyonTime changes from 10 to 0 (genesis time is %d)", genesisTime),
+			override1:         func(c *params.ChainConfig) { c.CanyonTime = uint64ptr(10) },
+			override2:         func(c *params.ChainConfig) { c.CanyonTime = uint64ptr(0) },
+			expectChainRewind: false,
+		},
+		{
+			name:              fmt.Sprintf("RegolithTime changes from 10 to 0 (genesis time is %d)", genesisTime),
+			override1:         func(c *params.ChainConfig) { c.RegolithTime = uint64ptr(10) },
+			override2:         func(c *params.ChainConfig) { c.RegolithTime = uint64ptr(0) },
+			expectChainRewind: false,
+		},
+		{
+			name:              fmt.Sprintf("ShanghaiTime changes from 10 to 0 (genesis time is %d)", genesisTime),
+			override1:         func(c *params.ChainConfig) { c.ShanghaiTime = uint64ptr(10) },
+			override2:         func(c *params.ChainConfig) { c.ShanghaiTime = uint64ptr(0) },
+			expectChainRewind: false,
+		},
+	}
+
+	for _, tc := range tcs {
+		t.Run(tc.name, func(t *testing.T) {
+			var genesis = &Genesis{
+				BaseFee:   big.NewInt(params.InitialBaseFee),
+				Config:    params.AllEthashProtocolChanges,
+				Timestamp: genesisTime,
+			}
+
+			// Prepare initial config for chain
+			tc.override1(genesis.Config)
+			db := rawdb.NewMemoryDatabase()
+			cc := DefaultCacheConfigWithScheme(rawdb.PathScheme)
+
+			// Start blockchain once to store config in DB
+			blockchain, _ := NewBlockChain(db, cc, genesis, nil, ethash.NewFaker(), vm.Config{}, nil, nil)
+
+			// Stop chain after 1 second
+			<-time.After(1 * time.Second)
+			blockchain.Stop()
+
+			// Setup a buffer to capture logs
+			logBuffer := bytes.Buffer{}
+			log.SetDefault(log.NewLogger(slog.NewTextHandler(&logBuffer, nil)))
+
+			// Restart chain with modified genesis config
+			tc.override2(genesis.Config)
+			blockchain, _ = NewBlockChain(db, cc, genesis, nil, ethash.NewFaker(), vm.Config{}, nil, nil)
+			<-time.After(1 * time.Second)
```

Comment:
same here, can we do some polling of some assertion with a shorter poll rate to make the test less flaky?

Reply:
In the end I removed this test entirely as discussed (we will rely on the more reliable / focussed unit tests for this PR).